### ISSUE TITLE
4.7.1 No recording ability

### ISSUE THREAD

**Issue by cgbur** [braden-w/whispering#219]
Just updated to 4.7.1 on mac intel. Unable to find any recording devices and lost my api keys so had to re add them. Still not able to find any devices. Restarted it several times and checked permissions.

I know its asking a lot but perhaps some level of automated testing could be added to CI, it is difficult to blindly trust updating currently without fear of the app being broken afterwards. 

I downgraded to 4.6 and it works as expected

**Comment by braden-w** [comment]
Oh man I'm so sorry about this, I cancelled 4.7.0 because it was broken, then made a few changes, double checked the app on my own machine, and it worked and I guess now it's broken once more :/ sorry about that and will revert!

**Comment by braden-w** [comment]
The problem was ever since I signed the application on MacOS, the audio permissions have been broken. Am trying to find a fix as we speak!

**Comment by braden-w** [comment]
@cgbur Could you confirm if the .dmgs in this release work?

https://github.com/braden-w/whispering/releases/tag/main

I tried replicating the issue but for some reason, it actually works on my machine.

Really strange, since I had the same issue before, made a modification to a .plist file, then undid it, but now the app always works for me despite having the older code 😅 unsure what changed, so just wanted confirm that the issue is on my machine .

**Comment by cgbur** [comment]
Totally understand testing UI based stuff on this many platforms is a nightmare. 

---

```
Error enumerating recording devices
Please make sure you have given permission to access your audio devices
```

But i have gone in and removed and added permissions

**Comment by braden-w** [comment]
Awesome, thanks so much! I'm going to continue iterating and testing and might send a few more .dmgs in prerelease your way since I can't get this bug on my computer anymore for some reason 😵‍💫

Did you try running `open /Applications/Whispering.app`? And by logging, do you mean logging with Console?

**Comment by braden-w** [comment]
The main issue is that this bug only exists in production builds from GitHub actions; it works in dev and prod builds on my machine. It has been a reproducibility nightmare hehe 😅

**Comment by cgbur** [comment]
Yeah trying 
```bash
open -a /Applications/Whispering.app -o ~/whiper.log
```
and 
```bash
tail -F ~/whisper.log
```

but the open command returns immediately no matter how i run it and does not hook in and give me any stdout or stderr logging streams.

**Comment by braden-w** [comment]
Ok, just like a few hours ago, I assume the issue is something to do with a .plist file being respected now that code signing is a thing. I'm reopening #218  and will be merging and setting a pre-release—will send you it once it's ready!

**Comment by braden-w** [comment]
@cgbur Could you take a look at this release? Took a while, but I tried removing sandboxing and hopefully it worked out!

https://github.com/braden-w/whispering/releases/tag/feature%2Fwhi-61-hotfix-infoplist-permissions-after-code-signing

**Comment by worldoptimizer** [comment]
![CleanShot 2024-07-26 at 13 30 23@2x](https://github.com/user-attachments/assets/b1b0002d-6132-4230-a756-da82b51455f8)

Using 4.7.1 I also get this. Hope I am not the cause for requesting the signing to begin with.

**Comment by cgbur** [comment]
Got the same double error, then this has remained:

![Screenshot 2024-07-26 at 9 11 20 AM](https://github.com/user-attachments/assets/d2fa392e-3235-4bbe-8208-6f715a824d86)

**Comment by braden-w** [comment]
Just pushed a fix in 4.7.2 that removes code signing and is a temporary fix. Will continue to experiment with plist for code signing!

**Comment by braden-w** [comment]
Going to close this and run experiments off the main branch to restore notarization in #223
